exporters/dsexporter/dsexporter.go

```diff
@@ -3,18 +3,33 @@ package main
 import (
 	"fmt"
 	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var pingCounter = prometheus.NewCounter(
+	prometheus.CounterOpts{
+		Name: "ping_request_count",
+		Help: "Number of requests handled by Ping handler",
+	},
 )
 
 func ping(w http.ResponseWriter, req *http.Request) {
+	if req.URL.Path != "/ping" {
+		// This handles unknown endpoints like 404 status code and an empty response body
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	pingCounter.Inc()
 	fmt.Fprintf(w, "pong")
 }
 
 func main() {
-	http.HandleFunc("/ping", ping)
+	prometheus.MustRegister(pingCounter)
 
-	fmt.Println("Starting dummy service on :8090")
-	err := http.ListenAndServe(":8090", nil)
-	if err != nil {
-		fmt.Println("Error starting dummy service:", err)
-	}
+	http.HandleFunc("/ping", ping)
```

Comment:
There shouldn't be a ping endpoint at this stage. We should now only listen to the metrics endpoint.